sql-ballerina/Module.md

```diff
@@ -0,0 +1,302 @@
+`sql:Client` to release the connection resources and avoid a connection leak as shown below.
+
+```ballerina
+stream<record{}, sql:Error> resultStream = 
+            dbClient->query("SELECT count(*) as total FROM students");
+
+record {|record {} value;|}|error? result = resultStream.next();
+
+if result is record {|record {} value;|} {
+    //valid result is returned.
+} else if result is error {
+    // An error is returned as the result.
+} else {
+    // Student table must be empty.
+}
+
+error? e = resultStream.close();
+```
+
+#### Updating Data
+
+This example demonstrates modifying data by executing an UPDATE statement via the `execute` remote function of 
+the client.
+
+```ballerina
+int age = 23;
+sql:ParameterizedQuery query = `UPDATE students SET name = 'John' 
+                                WHERE age = ${age}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Deleting Data
+
+This example demonstrates deleting data by executing a DELETE statement via the `execute` remote function of 
+the client.
+
+```ballerina
+string name = "John";
+sql:ParameterizedQuery query = `DELETE from students WHERE name = ${name}`;
+sql:ExecutionResult|sql:Error ret = check dbClient->execute(query);
+```
+
+#### Batch Updating Data
+
+This example demonstrates how to insert multiple records with a single INSERT statement that is executed via the 
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and 
+parameterized SQL query as same as the  above `execute` operations.
+
+```ballerina
+// Create the table with the records that need to be inserted.
+var data = [
+  { name: "John", age: 25  },
+  { name: "Peter", age: 24 },
+  { name: "jane", age: 22 }
+];
+
+// Do the batch update by passing the batches.
+sql:ParameterizedQuery[] batch = from var row in data
+                                 select `INSERT INTO students ('name', 'age')
+                                 VALUES (${row.name}, ${row.age})`;
+sql:ExecutionResult[] ret = check dbClient->batchExecute(batch);
+```
+
+#### Execute SQL Stored Procedures
+
+This example demonstrates how to execute a stored procedure with a single INSERT statement that is executed via the 
+`call` remote function of the client.
+
+```ballerina
+int uid = 10;
+sql:IntegerOutParameter insertId = new;
+
+sql:ProcedureCallResult|sql:Error ret = dbClient->call(`call InsertPerson(${uid}, ${insertId})`);
+if ret is error {
+    //An error returned
+} else {
+    stream<record{}, sql:Error>? resultStr = ret.queryResult;
+    if resultStr is stream<record{}, sql:Error> {
+        sql:Error? e = resultStr.forEach(function(record{} result) {
+        //can perform operations using 'result'.
+      });
+    }
+    check ret.close();
+}
+```
+Note that you have to explicitly invoke the close operation on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
```

Comment:
```suggestion
Note that you have to invoke the close operation explicitly on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
```

Reply:
Change Added in https://github.com/ballerina-platform/module-ballerina-sql/pull/228/commits/2656f12c62276e15a256c133465cb60304d0b041